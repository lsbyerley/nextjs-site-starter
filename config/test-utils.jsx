// test-utils.js
import { render } from '@testing-library/react';

// Add in any providers here if necessary:
// (ReduxProvider, ThemeProvider, etc)
const Providers = ({ children }) => {
  return children;
};

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export {
  default as userEvent,
  specialChars,
} from '@testing-library/user-event';
// export * from 'next-page-tester';

// override render method
export { customRender as render };
