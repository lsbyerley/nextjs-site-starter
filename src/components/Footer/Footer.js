import React from 'react';
import Link from 'next/link';

const Footer = () => {
  return (
    <footer className='py-6 mt-12 lg:mt-18 sm:pb-36 sm:py-12'>
      <div className='max-w-4xl px-4 mx-auto text-gray-800 dark:text-white'>
        <div className='pb-8 mb-2 border-t-2 border-gray-300 dark:border-white-300'></div>
        <div className='flex flex-col items-center justify-between lg:flex-row'>
          <p>Built with Next.js, Tailwind and Vercel</p>
          <div className='flex flex-wrap pt-2 space-x-2 font-medium sm:space-x-4 lg:pt-0'>
            <Link href='/about'>
              <a className={'transition-colors hover:text-yellow-500'}>About</a>
            </Link>
            <a
              href='#'
              className={'transition-colors hover:text-yellow-500'}
              target='_blank'
              rel='noreferrer'
            >
              Twitter
            </a>
            <a
              href='#'
              className={'transition-colors hover:text-yellow-500'}
              target='_blank'
              rel='noreferrer'
            >
              LinkedIn
            </a>
            <a
              href='#'
              className={'transition-colors hover:text-yellow-500'}
              target='_blank'
              rel='noreferrer'
            >
              GitHub
            </a>
            <a
              href='#'
              className={'transition-colors hover:text-yellow-500'}
              target='_blank'
              rel='noreferrer'
            >
              Hashnode
            </a>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
