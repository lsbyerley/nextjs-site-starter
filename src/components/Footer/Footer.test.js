import React from 'react';
import { render } from 'test-utils';
import Footer from './Footer';

const renderFooter = (props) => {
  const utils = render(<Footer {...props} />);
  const footer = utils.container.firstChild;
  return { ...utils, footer };
};

describe('Footer', () => {
  it('renders and matches snapshot', () => {
    const { footer } = renderFooter();
    expect(footer).toBeTruthy();
    expect(footer).toMatchSnapshot();
  });
});
