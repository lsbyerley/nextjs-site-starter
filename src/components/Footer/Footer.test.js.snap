// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders and matches snapshot 1`] = `
<footer
  class="py-6 mt-12 lg:mt-18 sm:pb-36 sm:py-12"
>
  <div
    class="max-w-4xl px-4 mx-auto text-gray-800 dark:text-white"
  >
    <div
      class="pb-8 mb-2 border-t-2 border-gray-300 dark:border-white-300"
    />
    <div
      class="flex flex-col items-center justify-between lg:flex-row"
    >
      <p>
        Built with Next.js, Tailwind and Vercel
      </p>
      <div
        class="flex flex-wrap pt-2 space-x-2 font-medium sm:space-x-4 lg:pt-0"
      >
        <a
          class="transition-colors hover:text-yellow-500"
          href="/about"
        >
          About
        </a>
        <a
          class="transition-colors hover:text-yellow-500"
          href="#"
          rel="noreferrer"
          target="_blank"
        >
          Twitter
        </a>
        <a
          class="transition-colors hover:text-yellow-500"
          href="#"
          rel="noreferrer"
          target="_blank"
        >
          LinkedIn
        </a>
        <a
          class="transition-colors hover:text-yellow-500"
          href="#"
          rel="noreferrer"
          target="_blank"
        >
          GitHub
        </a>
        <a
          class="transition-colors hover:text-yellow-500"
          href="#"
          rel="noreferrer"
          target="_blank"
        >
          Hashnode
        </a>
      </div>
    </div>
  </div>
</footer>
`;
